gemrb/core/Interface.cpp

```diff
@@ -250,6 +251,7 @@ Interface::Interface()
 	SpecialSpellsCount = -1;
 	SpecialSpells = NULL;
 	Encoding = "default";
+	SystemEncoding = nl_langinfo(CODESET);
```

Comment:
You're right that this isn't available on windows. How about using localeconv instead?

Reply:
I think localeconv don't have encoding information, it seems to have only information related to currency symbols, digit separators, and such. From what I quickly read, it will be required to use Windows-specific function, but I will investigate that more thoroughly later.

Reply:
maybe this would do, storing it before we override it:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/get-current-locale?view=vs-2019

Reply:
I've been reading a bit on the subject, and I'm afraid that I won't be able to add Windows support, mainly because I don't have Windows installed on my computer, so I cannot find solution with trial and error. As I understand, one can easily fetch code point, and iconv supports many of these. Windows also has native functions to convert from and to wide strings, which potentially could be used. https://docs.microsoft.com/en-us/windows/win32/intl/code-pages

My apologies, but could someone having Windows computer and expertise help me to finish this merge request?

Reply:
I suggest we just if-def it to UTF-16 in that case. I'm not sure about the defaults on windows and there's some problem, but it appears it is the 16 bit variant.

Also, is it really about the system encoding? You can have different encodings per partition on linux.

Reply:
Alright, we can do that. I will update review.

> Also, is it really about the system encoding? You can have different encodings per partition on linux.

Well, usually yes. As I can read Gnome's Glib assumes that filenames are encoded in UTF-8, and Qt assumes they are in user encoding. Tools like `ls` also seem to depend on user-locale, `LANG=c ls` doesn't show correctly filenames with non-ASCII characters on my system. I suppose assuming that filenames are encoded in user-locale is reasonable thing to do.